packages/nimble/inst/include/nimble/EigenTypedefs.h

```diff
@@ -85,22 +85,69 @@ nimSmartPtr<EIGEN_SVDCLASS_R>   EIGEN_SVD_R(const Eigen::MatrixBase<Derived> &x,
 }
 
 template<class Derived>
-void EIGEN_EIGEN_INTERNAL(const Eigen::MatrixBase<Derived> &x, bool valuesOnly, EIGEN_EIGENCLASS_R *returnClass) {
-  // nimSmartPtr<EIGEN_EIGENCLASS> returnClass = new EIGEN_EIGENCLASS;
-  returnClass->getValues().initialize(0, 0, x.rows());
+void EIGEN_EIGEN_INTERNAL(const Eigen::MatrixBase<Derived> &x,  bool symmetric, bool valuesOnly, EIGEN_EIGENCLASS_R *returnClass) {
+    returnClass->getValues().initialize(0, 0, x.rows());
 	Map<VectorXd> Eig_eigVals(returnClass->getValues().getPtr(),x.rows());
+	if(!valuesOnly){
+		returnClass->getVectors().initialize(0, 0, x.rows(), x.cols());
+	}
 	if(x.rows() != x.cols()) {
 	 _nimble_global_output <<"Run-time size error: expected matrix argument to eigen() to be square."<<"\n"; nimble_print_to_R(_nimble_global_output);
 	}
     Eigen::DecompositionOptions eigOpts = valuesOnly ? EigenvaluesOnly : ComputeEigenvectors;
-	SelfAdjointEigenSolver<MatrixXd> solver(x, eigOpts); // The MatrixXd here doesn't seem generic, but I couldn't get it to work otherwise and it would be odd to do an Eigen decomposition on anything else. -Perry
-	Eig_eigVals = solver.eigenvalues().reverse();
-	if(!valuesOnly){
-	  returnClass->getVectors().initialize(0, 0, x.rows(), x.cols());
-	  Map<MatrixXd> Eig_eigVecs(returnClass->getVectors().getPtr(),x.rows(),x.cols());
-	  Eig_eigVecs = solver.eigenvectors().rowwise().reverse();	
+	if(!symmetric){
+		symmetric = true;
```

Comment:
nit: The symmetry check would cleaner as a helper function. E.g. you could `return false` rather than using two break statements. It would improve readability of `EIGEN_EIGEN_INTERNAL()`.